### PR TITLE
Duplicates in addrsets

### PR DESCRIPTION
**Pull request by trozet**
**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by dougsland** [comment]
lgtm

File: go-controller/pkg/ovn/address_set/address_set.go [hunk 1]
```diff
@@ -683,13 +682,11 @@ func splitIPsByFamily(ips []net.IP) (v4 []net.IP, v6 []net.IP) {
...
 
-func ipToStringSort(ips []net.IP) []string {
+func ipsToString(ips []net.IP) []string {
 	// my kingdom for a ".values()" function
 	out := make([]string, 0, len(ips))
 	for _, ip := range ips {
 		out = append(out, ip.String())
 	}
-	// so the tests are predictable
-	sort.Strings(out)
```
Comment (tssurya) [inline go-controller/pkg/ovn/address_set/address_set.go:693]: ++

File: go-controller/pkg/ovn/address_set/address_set.go [hunk 2]
```diff
@@ -616,16 +611,13 @@ func (as *ovnAddressSet) addIPs(ips []net.IP) ([]ovsdb.Operation, error) {
 
 // deleteIPs removes selected IPs from the existing address_set
 func (as *ovnAddressSet) deleteIPs(ips []net.IP) ([]ovsdb.Operation, error) {
-	// dedup
-	uniqIPs := make([]string, 0, len(ips))
```
Comment (tssurya) [inline go-controller/pkg/ovn/address_set/address_set.go:621]: ha its weird we were running this loop and then checking if the list was empty :D

File: go-controller/pkg/ovn/address_set/address_set.go [hunk 3]
```diff
@@ -682,11 +674,11 @@ func splitIPsByFamily(ips []net.IP) (v4 []net.IP, v6 []net.IP) {
 	return
 }
 
-func ipsToString(ips []net.IP) []string {
-	// my kingdom for a ".values()" function
-	out := make([]string, 0, len(ips))
+// Takes a slice of IPs and returns a slice with unique IPs
+func ipsToStringUnique(ips []net.IP) []string {
```
Comment (tssurya) [inline go-controller/pkg/ovn/address_set/address_set.go:678]: I agree this is much safer, curious if we hit the duplicate issue for address sets as well? where did we see this?

File: go-controller/pkg/ovn/address_set/address_set_test.go [hunk 4]
```diff
@@ -505,6 +505,41 @@ var _ = ginkgo.Describe("OVN Address Set operations", func() {
...
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				asFactory = NewOvnAddressSetFactory(libovsdbOvnNBClient)
+
+				_, err = config.InitConfig(ctx, nil, nil)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				as, err := asFactory.NewAddressSet("foobar", nil)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				ops, err := as.AddIPsReturnOps([]net.IP{net.ParseIP(addr1), net.ParseIP(addr1)})
```
Comment (tssurya) [inline go-controller/pkg/ovn/address_set/address_set_test.go:524]: ++ nice

**Review by tssurya — APPROVED** [review]
/lgtm

File: go-controller/pkg/ovn/address_set/address_set.go (hunk 1 above)
Comment (dcbw) [reply go-controller/pkg/ovn/address_set/address_set.go:693]: Yeah, this was only needed for ovn-nbctl when we compared command arguments directly...

**Pull request merged**